File: fixtures/python/operations.py
"""Pure arithmetic operations."""

from errors import DivisionByZeroError


def add(a: float, b: float) -> float:
    """Return the sum of a and b."""
    return a + b


def subtract(a: float, b: float) -> float:
    """Return the difference of a and b."""
    return a - b


def multiply(a: float, b: float) -> float:
    """Return the product of a and b."""
    return a * b


def divide(a: float, b: float) -> float:
    """Return the quotient of a and b.

    Raises:
        DivisionByZeroError: If b is zero.
    """
    if b == 0:
        raise DivisionByZeroError
    return a / b

File: fixtures/dart/lib/operations.dart
import 'errors.dart';

/// Pure arithmetic operations.
class Operations {
  /// Returns the sum of [a] and [b].
  static double add(double a, double b) {
    return a + b;
  }

  /// Returns the difference of [a] and [b].
  static double subtract(double a, double b) {
    return a - b;
  }

  /// Returns the product of [a] and [b].
  static double multiply(double a, double b) {
    return a * b;
  }

  /// Returns the quotient of [a] and [b].
  ///
  /// Throws [DivisionByZeroError] if [b] is zero.
  static double divide(double a, double b) {
    if (b == 0.0) {
      throw DivisionByZeroError();
    }
    return a / b;
  }
}
